packages/multi-test/src/wasm.rs

```diff
@@ -82,29 +83,33 @@ where
         api: &dyn Api,
         contract_addr: Addr,
         storage: &mut dyn Storage,
-        router: &Router<C>,
+        router: &Router<ExecC, QueryC>,
         block: &BlockInfo,
         msg: Vec<u8>,
     ) -> AnyResult<AppResponse>;
 }
 
-pub struct WasmKeeper<C> {
+pub struct WasmKeeper<ExecC = Empty, QueryC = Empty> {
     /// code is in-memory lookup that stands in for wasm code
     /// this can only be edited on the WasmRouter, and just read in caches
-    codes: HashMap<usize, Box<dyn Contract<C>>>,
+    codes: HashMap<usize, Box<dyn Contract<ExecC>>>,
+    /// Just marker to make type elision fork when using it as `Wasm` trait
```

Comment:
Actually, the contract should have QueryC in it's type.

Update: I looked at this code (also in cosmwasm-std) and it seems the query types are not enforced in DepsMut (which is where they are used). This is proper here, but maybe we can look at making cosmwasm-std more type safe

(TODO: @ethanfrey make an issue on cosmwasm-std)